src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -58,6 +58,53 @@ public void GetProcessesByName_RemoteMachineNameUnix_ThrowsPlatformNotSupportedE
             Assert.Throws<PlatformNotSupportedException>(() => Process.GetProcessesByName(currentProcess.ProcessName, machineName));
         }
 
+        [Fact]
+        public void GetProcesses_LongProcessName()
+        {
+            string commandName = "sleep";
+
+            // sleep program doesn't exist on some distros
+            if (IsProgramInstalled(commandName))
+            {
+                string tmpDir = Path.Combine(TestDirectory, GetTestFileName());
+                try
+                {
+                    // create tmp folder and copy sleep command inside it renamed
+                    Directory.CreateDirectory(tmpDir);
+                    string longProcessName = "123456789012345678901234567890";
+                    string sleepCommandPathFileName = Path.Combine(tmpDir, longProcessName);
+                    File.Copy(GetProgramPath(commandName), sleepCommandPathFileName);
+                    Assert.True(File.Exists(sleepCommandPathFileName));
+
+                    // start sleep program and wait for some seconds
```

Comment:
We typically use 30sec for delays that are essentially timeouts. To be sure they are long enough